store/sqlstore/post_store.go

```diff
@@ -345,7 +345,7 @@ func (s *SqlPostStore) GetFlaggedPosts(userId string, offset int, limit int) (*m
 	pl := model.NewPostList()
 
 	var posts []*model.Post
-	if _, err := s.GetReplica().Select(&posts, "SELECT *, (SELECT count(Posts.Id) FROM Posts WHERE Posts.RootId = (CASE WHEN p.RootId = '' THEN p.Id ELSE p.RootId END) AND Posts.DeleteAt = 0) as ReplyCount FROM Posts p WHERE Id IN (SELECT Name FROM Preferences WHERE UserId = :UserId AND Category = :Category) AND DeleteAt = 0 ORDER BY CreateAt DESC LIMIT :Limit OFFSET :Offset", map[string]interface{}{"UserId": userId, "Category": model.PREFERENCE_CATEGORY_FLAGGED_POST, "Offset": offset, "Limit": limit}); err != nil {
+	if _, err := s.GetReplica().Select(&posts, "SELECT p.*, COALESCE(T.ReplyCount, 0) as ReplyCount FROM Posts p LEFT JOIN Threads as T ON T.PostId = (CASE WHEN p.RootId IS NULL OR p.RootId = '' THEN p.Id ELSE p.RootId END) WHERE p.Id IN (SELECT Name FROM Preferences WHERE UserId = :UserId AND Category = :Category) AND p.DeleteAt = 0 ORDER BY p.CreateAt DESC LIMIT :Limit OFFSET :Offset", map[string]interface{}{"UserId": userId, "Category": model.PREFERENCE_CATEGORY_FLAGGED_POST, "Offset": offset, "Limit": limit}); err != nil {
```

Comment:
@jespino can RootID be null? I think that condition can be simpler.

Reply:
Yes from the database perspective, I guess no from the application perspective. I was playing safe there. :)